orchestration/dagster_orchestration/hca_manage/tests/test_hca_manage.py

```diff
@@ -0,0 +1,190 @@
+        with patched_soft_delete as mock_submit, patched_put_in_bucket as mock_bucket_upload:
+            with NamedTemporaryFile() as fake_file:
+                returned_job_id = self.manager.soft_delete_rows(fake_file.name, 'table_a')
+            mock_bucket_upload.assert_called_once_with(
+                local_file=ObjectOfType(BufferedIOBase),
+                target_table='table_a',
+            )
+            mock_submit.assert_called_once_with(target_table='table_a', target_path=fake_google_path)
+            self.assertEqual(returned_job_id, fake_job_id)
+
+    def test_get_dataset_id_gets_id_of_first_dataset(self):
+        enumerate_datasets_response = Mock()
+        first_dataset = Mock()
+        first_dataset.id = 'abc'
+
+        second_dataset = Mock()
+        second_dataset.id = 'def'
+        enumerate_datasets_response.items = [first_dataset, second_dataset]
+        self.manager.data_repo_client.enumerate_datasets.return_value = enumerate_datasets_response
+        self.assertEqual(self.manager.get_dataset_id(), 'abc')
+
+    def test_submit_soft_delete_submits_dataset_deletion_with_expected_params(self):
+        enumerate_datasets_response = Mock()
+        single_dataset = Mock()
+        single_dataset.id = 'abc'
+        enumerate_datasets_response.items = [single_dataset]
+        self.manager.data_repo_client.enumerate_datasets.return_value = enumerate_datasets_response
+        self.manager._submit_soft_delete('table_a', 'bad_rows_a.csv')
+        self.manager.data_repo_client.apply_dataset_data_deletion.assert_called_once_with(
+            id='abc',
+            data_deletion_request=DataDeletionRequest(
+                delete_type="soft",
+                spec_type="gcsFile",
+                tables=[
+                    {
+                        "gcsFileSpec": {
+                            "fileType": "csv",
+                            "path": 'bad_rows_a.csv',
+                        },
+                        "tableName": 'table_a'
+                    }
+                ]
+            )
+        )
+
+    def test_submit_snapshot_request_ignores_qualifier_if_not_present(self):
+        self.manager.submit_snapshot_request(qualifier=None)
+        self.manager.data_repo_client.create_snapshot.assert_called_once_with(
+            snapshot=ObjectWithAttributes(
+                SnapshotRequestModel,
+                name=StringMatchingRegex(r'datasetname___\d{8}'))
+        )
+
+    def test_submit_snapshot_request_uses_qualifier_if_present(self):
+        self.manager.submit_snapshot_request(qualifier='steve')
+        self.manager.data_repo_client.create_snapshot.assert_called_once_with(
+            snapshot=ObjectWithAttributes(
+                SnapshotRequestModel,
+                name=StringMatchingRegex(r'datasetname___\d{8}_steve'))
+        )
+
+    def test_delete_snapshot_fetches_id_if_missing(self):
+        enumerate_snapshots_response = Mock()
```

Comment:
General nit: for readability please try to structure your tests like the following:

```
<preamble/setup>

<invocation of unit under test>

<assertions>
```

so in this case:

```suggestion
        enumerate_snapshots_response = Mock()
        single_snapshot = Mock()
        single_snapshot.id = 'abc'
        enumerate_snapshots_response.items = [single_snapshot]
        self.manager.data_repo_client.enumerate_snapshots.return_value = enumerate_snapshots_response
        
        self.manager.delete_snapshot(snapshot_name='steve')
        
        self.manager.data_repo_client.delete_snapshot.assert_called_once_with('abc')
```